### PR TITLE
feat(cli) better version information for dependency binaries

### PR DESCRIPTION
**Pull request by Tieske**
Replaces #1221

### CODE REVIEW

File: kong/cmd/utils/nginx_signals.lua [hunk 1]
```diff
@@ -2,22 +2,31 @@ local pl_utils = require "pl.utils"
...
-    log.debug("%s: '%s'", cmd, v_str:sub(1, -2))
-    return v_str:match "^nginx version: ngx_openresty/" or
-           v_str:match "^nginx version: openresty/"
+  local cmd = fmt("%s %s", bin_path, nginx_version_command)
+  local ok, _, _, stderr = pl_utils.executeex(cmd)
+  if ok and stderr then
+    log.debug("%s: '%s'", cmd, stderr:sub(1, -2))
+    local version_match = stderr:match(nginx_version_pattern)
+    if (not version_match) or (not nginx_compatible:matches(version_match)) then
+      return nil, "Incompatible nginx found. Kong requires OpenResty, version "..tostring(nginx_compatible) ..
```
Comment (thibaultCha) [inline kong/cmd/utils/nginx_signals.lua]: Please do not capitalize error messages here to stay consistent with other errors in the CLI and respect the punctuation when thy are printed to the user ("Error: ...").
Ditto for the other 2-3 occurrences in this changelog

Comment (thibaultCha) [inline kong/cmd/utils/nginx_signals.lua]: Also: the shorter the better

**Comment by Tieske** [comment]
de-capitalized

**Comment by thibaultCha** [comment]
Could you rebase your changes on the latest `refactor/cli`?

**Pull request merged**